terragrunt/aws/cloudfront/waf.tf

```diff
@@ -0,0 +1,183 @@
+      metric_name                = "APIRateLimit"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "AWSManagedRulesCommonRuleSet"
+    priority = 30
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesCommonRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesCommonRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "AWSManagedRulesKnownBadInputsRuleSet"
+    priority = 40
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesKnownBadInputsRuleSet"
+        vendor_name = "AWS"
+
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesKnownBadInputsRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "AWSManagedRulesLinuxRuleSet"
+    priority = 50
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesLinuxRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesLinuxRuleSet"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "api"
+    sampled_requests_enabled   = false
+  }
+}
+
+resource "aws_cloudwatch_log_group" "simplify_privacy_statements_waf" {
+  name              = "/aws/kinesisfirehose/simplify_privacy_statements_waf"
+  retention_in_days = 14
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "simplify_privacy_statements_waf" {
+  provider = aws.us-east-1
+
+  name        = "aws-waf-logs-simplify_privacy"
+  destination = "extended_s3"
+
+  server_side_encryption {
+    enabled = true
+  }
+
+  extended_s3_configuration {
+    role_arn           = aws_iam_role.waf_log_role.arn
+    prefix             = "waf_acl_logs/AWSLogs/154232573588/"
+    bucket_arn         = "arn:aws:s3:::${var.billing_code}-tf"
+    compression_format = "GZIP"
+
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = aws_cloudwatch_log_group.simplify_privacy_statements_waf.name
+      log_stream_name = "WAFLogS3Delivery"
+    }
+  }
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_wafv2_web_acl_logging_configuration" "simplify_privacy_statements_waf" {
+  provider                = aws.us-east-1
+  log_destination_configs = [aws_kinesis_firehose_delivery_stream.simplify_privacy_statements_waf.arn]
+  resource_arn            = aws_wafv2_web_acl.simplify_privacy_statements_waf.arn
+}
+
+resource "aws_iam_role" "waf_log_role" {
```

Comment:
You'll also need to add an IAM policy that lets this role write to the target S3 bucket.  It'll look something like this:/
https://github.com/cds-snc/scan-websites/blob/bee89f1853a5ea9f90b10fd6a1146b43f5d924e9/terragrunt/aws/api/iam.tf#L208-L225

The only gotcha is that the S3 bucket doesn't exist yet since I haven't had a chance to create the Cloud Based Sensor PR that creates it 😅